src/frontend/src/lib/components/registry/teams/TeamTournamentHistory.svelte

```diff
@@ -0,0 +1,154 @@
+<script lang="ts">
+  import { page } from '$app/stores';
+  import { user } from '$lib/stores/stores';
+  import { onMount } from 'svelte';
+  import type { Team } from '$lib/types/team';
+  import type { UserInfo } from '$lib/types/user-info';
+  import type { ProfileTournamentPlacement } from '$lib/types/tournament-placement';
+  import Section from '$lib/components/common/Section.svelte';
+  import Table from '$lib/components/common/Table.svelte';
+  import logo from '$lib/assets/logo.png';
+  import Button from '$lib/components/common/buttons/Button.svelte';
+  import LL from '$i18n/i18n-svelte';
+  import GameModeSelect from '$lib/components/common/GameModeSelect.svelte';
+  import PlayerName from '$lib/components/tournaments/registration/PlayerName.svelte';
+
+  let user_info: UserInfo;
+
+  user.subscribe((value) => {
+    user_info = value;
+  });
+
+  export let team: Team;
+
+  let mode: string | null = null;
+  let game: string | null = null;
+  let from: string | null = null;
+  let to: string | null = null;
+  let team_placements: ProfileTournamentPlacement[] = [];
+  let filtered_team_placements: ProfileTournamentPlacement[] = [];
+  // Default 'silver' from PlacementsDisplay.svelte is less readable than I'd like
+  let podium_style: { [key: number]: string } = { 1: 'gold', 2: 'bg-slate-400/60', 3: 'bronze' };
+
+  function toDate(unix_timestamp: number) {
+    return new Date(unix_timestamp * 1000).toLocaleDateString();
+  }
+
+  async function fetchData() {
+    // API
+    let url = `/api/tournaments/teams/placements/${team.id}`;
+    const res = await fetch(url);
+    if (res.status !== 200) {
+      return;
+    }
+    let body = await res.json();
+    team_placements = body.tournament_team_placements;
+
+    // Filtering
+    filtered_team_placements = [...team_placements];
+
+    if (game) {
+      filtered_team_placements = filtered_team_placements.filter((item) => item.game === game);
+    }
+    if (mode) {
+      filtered_team_placements = filtered_team_placements.filter((item) => item.mode === mode);
+    }
+    if (to) {
+      filtered_team_placements = filtered_team_placements.filter((item) => {
+        return item.date_end <= Date.parse(to) / 1000;
+      });
+    }
+
+    filtered_team_placements = filtered_team_placements.sort((a, b) => {
+      return b.date_start - a.date_start;
+    });
```

Comment:
If we are filtering on the frontend I'd argue that this should be moved into a separate function for filtering and then the filter button calls that, placements shouldn't change often enough that we need to make a new API call every time we're filtering the placements.

Reply:
Done. 👍